File: ChainOfResponsibilityPatternExample/src/main/java/SecuredFilter.java
public class SecuredFilter extends RequestFilter {
    @Override
    public void handleRequest(final Request request) {
        System.out.println("Running Secured Filter...");

        if (!Helper.isSecured(request)) {
            System.out.println(Helper.getResource());
            return;
        }
        successor.handleRequest(request);
    }
}

File: ChainOfResponsibilityPatternExample/src/main/cpp/src/secured_filter.cpp
#include "secured_filter.h"
#include <iostream>

void SecuredFilter::handleRequest(int request){
  switch(request){
    case 2:
      std::cout << "Request being handled by SecuredFilter" << std::endl;
      break;
    default:
      this->successor->handleRequest(request);
  }
}
